File: Java/src/LinkedList/HandsOn/LinkedList.java
package LinkedList.HandsOn;

public class LinkedList<T extends Comparable<T>> {
    LinkedList() {
    }

    Node<T> head = null;

    void addNode(T data) {
        if (head == null) {
            head = new Node<>(data);
        } else {
            Node curr = head;
            while (curr.getNext() != null) {
                curr = curr.getNext();
            }
            curr.setNext(new Node<T>(data));
        }
    }

    void display() {
        if (head == null) {
            System.out.println("Empty Linked List");
        } else {
            Node curr = head;
            int i = 0;
            while (curr != null) {
                System.out.println(i + " Node is having value: " + curr.getData() + " ");
                curr = curr.getNext();
                i++;
            }

        }
    }
}

File: Python/DS/linked_list.py
class Node:
    # Function to initialise the node object
    def __init__(self, data):
        self.data = data  # Assign data 
        self.next = None  # Initialize next as null

    def get_next(self):
        return self.next

    def set_next(self, node):
        self.next = node

    def get_data(self):
        return self.data


# Linked List class contains a Node object 
class LinkedList:
    # Function to initialize head 
    def __init__(self):
        self.head = None

    def add_next(self, data):
        if not self.head:
            print("first")
            self.head = Node(data)
        else:
            curr = self.head
            print("second")
            while curr.get_next():
                curr = curr.get_next()
            curr.set_next(Node(data))

    def print(self):
        if not self.head:
            print("Empty")
        else:
            curr = self.head
            i = 1
            while curr:
                print(f"Node[{i}]=", curr.data, end=" ")
                i += 1
                curr = curr.get_next()


# Node creation
if __name__ == '__main__':
    list = LinkedList()
    list.print()
    list.add_next(2)
    list.add_next(5)
    list.print()
